lib/mail_room/mailbox.rb

```diff
@@ -30,45 +35,52 @@ module MailRoom
       :ssl => true,
       :start_tls => false,
       :delete_after_delivery => false,
-      :delivery_options => {}
+      :delivery_options => {},
+      :arbitration_method => 'noop',
+      :arbitration_options => {}
     }
 
     # Store the configuration and require the appropriate delivery method
     # @param attributes [Hash] configuration options
     def initialize(attributes={})
       super(*DEFAULTS.merge(attributes).values_at(*members))
-
-      require_relative("./delivery/#{(delivery_method)}")
     end
 
-    # move to a mailbox deliverer class?
     def delivery_klass
-      case delivery_method
-      when "noop"
-        Delivery::Noop
-      when "logger"
-        Delivery::Logger
-      when "letter_opener"
-        Delivery::LetterOpener
-      when "sidekiq"
-        Delivery::Sidekiq
-      when "que"
-        Delivery::Que
-      else
-        Delivery::Postback
-      end
+      Delivery[delivery_method]
+    end
+
+    def arbitration_klass
+      Arbitration[arbitration_method]
+    end
+
+    def delivery
+      @delivery ||= delivery_klass.new(parsed_delivery_options)
+    end
+
+    def arbitrator
+      @arbitrator ||= arbitration_klass.new(parsed_arbitration_options)
+    end
+
+    def deliver?(uid)
+      arbitrator.deliver?(uid)
     end
 
     # deliver the imap email message
     # @param message [Net::IMAP::FetchData]
     def deliver(message)
-      message = message.attr['RFC822']
-      return true unless message
-      
-      delivery_klass.new(parsed_delivery_options).deliver(message)
+      body = message.attr['RFC822']
```

Comment:
In IMAP land the body appears to be the payload, with headers and all:

```
    RFC822
         Functionally equivalent to BODY[], differing in the syntax of
         the resulting untagged FETCH data (RFC822 is returned).
```

Per https://tools.ietf.org/html/rfc3501